react-native-hub-app/src/checklist.tsx

```diff
@@ -0,0 +1,482 @@
+const USER_THREAD_ID = 'user_thread-' + version;
+const sleep = (m) => new Promise((r) => setTimeout(r, m));
+
+interface StateProps {
+  steps: any,
+  step: number,
+  errorMessage: string,
+  threadId?: ThreadID,
+  ctx?: Context,
+  db?: Client,
+  entityId?: string,
+  bucketUrl?: string
+}
+class CheckList extends React.Component<StateProps> {
+  constructor(props) {
+    super(props);
+  }
+
+  // you could also do this, so no constructor needed
+  state: StateProps = {
+    steps: [
+      {key: 'Step 0', name: 'Prepare Identity & API Token', status: 0},
+      {key: 'Step 1', name: 'Setup ThreadDB', status: 0},
+      {key: 'Step 2', name: 'Add Instance to Collection', status: 0},
+      {key: 'Step 3', name: 'Query from our Collection', status: 0},
+      {key: 'Step 4', name: 'Push webpage to User Bucket', status: 0},
+    ],
+    step: 0,
+    errorMessage: '',
+  };
+
+  incrementStatus(index: number) {
+    const steps = this.state.steps;
+    const data = steps[index];
+    if (data.status >= MAX_STEPS) {
+      return;
+    }
+    data.status = data.status += 1;
+    steps[index] = data;
+    this.setState({
+      steps: steps,
+    });
+  }
+
+  async generateIdentity(): Promise<Libp2pCryptoIdentity> {
+    let idStr = await AsyncStorage.getItem(IDENTITY_KEY);
+    if (idStr) {
+      return await Libp2pCryptoIdentity.fromString(idStr);
+    } else {
+      const id = await Libp2pCryptoIdentity.fromRandom();
+      idStr = id.toString();
+      await AsyncStorage.setItem(IDENTITY_KEY, idStr);
+      return id;
+    }
+  }
+
+  async getUserToken(database, id): Promise<string> {
+    let token;// = await AsyncStorage.getItem(TOKEN_KEY);
+    if (token) {
+      return token;
+    }
+    token = await database.getToken(id);
+    await AsyncStorage.setItem(TOKEN_KEY, token);
+    return token;
+  }
+
+  async getContext(): Promise<Context> {
+    // Pull the stored context to reuse if available && valid date
+    let contextStr = await AsyncStorage.getItem(CONTEXT_KEY);
+    if (contextStr) {
+      const ctxJson = JSON.parse(contextStr);
+      if (
+        ctxJson['x-textile-api-sig-msg'] && (Date.parse(ctxJson['x-textile-api-sig-msg'])) > (new Date()).getTime()) {
+        // Not expired
+        const ctx = Context.fromJSON(ctxJson, API_URL);
+        return ctx;
+      }
+    }
+  }
+
+  async getUserThread(db: Client, ctx: Context): Promise<ThreadID> {
+    let idStr = await AsyncStorage.getItem(USER_THREAD_ID);
+    if (idStr) {
+      /**
+       * Temporary hack to get ThreadID working in RN
+       */
+      const temp: ThreadID = ThreadID.fromString(idStr);
+      // @ts-ignore
+      const id = ThreadID.fromBytes(Buffer.from(temp.buf));
+      return id;
+    } else {
+      const id: ThreadID = ThreadID.fromRandom();
+      await AsyncStorage.setItem(USER_THREAD_ID, id.toString());
+
+      /**
+       * Each new ThreadID requires a `newDB` call.
+       */
+      await db.newDB(id, ctx.withThread(id))
```

Comment:
ctx.withThread is no longer needed here, it is called automatically by the client upon creation of the new thread for the db.